gef.py

```diff
@@ -3716,41 +3717,55 @@ def parsed_arglist(arglist):
 
     def synchronize(self):
         """Submit all active breakpoint addresses to IDA/BN"""
-        breakpoints = gdb.breakpoints() or []
-        old_bps = []
 
+        pc = current_arch.pc
+
+        vmmap = get_process_maps()
+        base_address = min([x.page_start for x in vmmap if x.path == get_filepath()])
+        end_address = max([x.page_end for x in vmmap if x.path == get_filepath()])
+        if pc > end_address or pc < base_address:
+            # do not sync in library
+            return
+
+        breakpoints = gdb.breakpoints() or []
+        gdb_bps = []
         for x in breakpoints:
             if x.enabled and not x.temporary:
-                val = gdb.parse_and_eval(x.location)
-                addr = str(val).strip().split()[0]
-                addr = long(addr, 16)
-                old_bps.append(addr)
+                addr = long(gdb.parse_and_eval(x.location).address)
+                if addr > end_address or addr < base_address:
+                    continue
+                gdb_bps.append(addr-base_address)
+
+        added = set(gdb_bps) - set(self.old_bps)
```

Comment:
Why not have these all be sets (self.old_bps, gdb_bps, etc) and only convert to a list when you need them? Do they ever need to be a list?